lib/webdriveragent.js

```diff
@@ -137,6 +147,28 @@ class WebDriverAgent {
     }
   }
 
+  async updateProjectFile () {
+    let projectFilePath = `${this.agentPath}/${PROJECT_FILE}`;
+    try {
+      // backup the file, and then update the bundle id for the runner
+      await fs.copyFile(projectFilePath, `${projectFilePath}.old`);
+      let changedFiles = await replaceInFile({
+        files: projectFilePath,
+        replace: WDA_RUNNER_BUNDLE_ID,
+        with: this.updatedWDABundleId,
+      });
+      log.debug(`Successfully updated '${changedFiles.join(', ')}' with bundle id '${this.updatedWDABundleId}'`);
+    } catch (err) {
+      log.debug(`Error updating project file: ${err.message}`);
+      log.warn(`Unable to update project file '${projectFilePath}' with ` +
+               `bundle id '${this.updatedWDABundleId}'. WebDriverAgent may not start`);
+    }
+  }
+
+  async resetProjectFile () {
+
```

Comment:
Ah! This was a thought, and one that needs discussing. Should we return the state, do you think? As it stands the backup of the project file quickly (after the second run) becomes the modified version. I'm not sure that this matters, and am happy to do either option (adding reset, or not removing this stub).